### PR TITLE
Replace test stages with ruby scripts

### PR DESCRIPTION
**Pull request by brunoOSI**
### Jira Story

No story

## Description
This fixes the issue with false positives with the unified test stage.
It also ports all the logic to run testing and store reporting to more manageable ruby scripts.

### CODE REVIEW

File: scripts/ci/karma_test.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+test_args = "-p intake_accelerator#{ENV['BUILD_NAME']}_test -f docker/test/docker-compose.yml"
+exec("docker-compose #{test_args} run run_tests bin/ci-javascript")
```
Comment (NagaPradeepGajula-osi) [inline scripts/ci/karma_test.rb]: isn't this supposed to be bin/karma?

Comment (brunoOSI) [reply scripts/ci/karma_test.rb]: No, it has to be that script to run headless:
```
Xvfb :99 -ac +extension RANDR -screen 0 2500x2500x16 &
DISPLAY=:99 bin/karma
```

**Comment by brunoOSI** [comment]
retest this please

**Comment by brunoOSI** [comment]
retest this please

**Pull request merged**